app/api/tutors/new/route.js

```diff
@@ -0,0 +1,21 @@
+import { db } from "@/firebase";
+
+export async function POST(request) {
+  const { userID, gender, bookingLink, ageGroup, subjects } =
+    await request.json();
+  let status = "success";
+  let tutor = {
+    gender: gender,
+    bookingLink: bookingLink,
+    ageGroup: ageGroup,
+    subjects: subjects,
+  };
+  try {
+    db.collection("tutors").doc(userID).set({ tutor });
+    return Response.json({ status, data: tutor });
+  } catch (error) {
+    status = "Error";
+    const message = `Error adding document: ${error}`;
+    return Response.json({ status, message });
+  }
```

Comment:
Hey @egwilborn the `error` here could be an object which will cause another error when you pass it as is as a response. Usually, we stringify the error before passing it to the client. Here's my suggestion for this block of code.

```suggestion
  } catch (error) {
    return Response.json({ status: "error", message: JSON.stringify(error) });
  }
```